### ISSUE TITLE
Error in bam2gdf

### ISSUE THREAD

**Issue by kokyriakidis** [sbslee/pypgx#4]
I am getting the following error when I run bam2gdf in gstt1:
```
[INFO] PyPGx v0.1.22
[INFO] Command:
[INFO]     /Users/kokyriakidis/Documents/GitHub/Pharmakon/.snakemake/conda/3247b051/bin/pypgx bam2gdf hg38 gstp1 vdr /Users/kokyriakidis/Desktop/test_folder/KOSTAS/genes/gstp1/gdf/gstp1.gdf /Users/kokyriakidis/Desktop/test_folder/KOSTAS/applybqsr/KOSTAS.bam
[INFO] Sample IDs: ['KOSTAS']
[INFO] Contigs: ['1', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '2', '20', '21', '22', '3', '4', '5', '6', '7', '8', '9', 'MT', 'X', 'Y', 'KI270728.1', 'KI270727.1', 'KI270442.1', 'KI270729.1', 'GL000225.1', 'KI270743.1', 'GL000008.2', 'GL000009.2', 'KI270747.1', 'KI270722.1', 'GL000194.1', 'KI270742.1', 'GL000205.2', 'GL000195.1', 'KI270736.1', 'KI270733.1', 'GL000224.1', 'GL000219.1', 'KI270719.1', 'GL000216.2', 'KI270712.1', 'KI270706.1', 'KI270725.1', 'KI270744.1', 'KI270734.1', 'GL000213.1', 'GL000220.1', 'KI270715.1', 'GL000218.1', 'KI270749.1', 'KI270741.1', 'GL000221.1', 'KI270716.1', 'KI270731.1', 'KI270751.1', 'KI270750.1', 'KI270519.1', 'GL000214.1', 'KI270708.1', 'KI270730.1', 'KI270438.1', 'KI270737.1', 'KI270721.1', 'KI270738.1', 'KI270748.1', 'KI270435.1', 'GL000208.1', 'KI270538.1', 'KI270756.1', 'KI270739.1', 'KI270757.1', 'KI270709.1', 'KI270746.1', 'KI270753.1', 'KI270589.1', 'KI270726.1', 'KI270735.1', 'KI270711.1', 'KI270745.1', 'KI270714.1', 'KI270732.1', 'KI270713.1', 'KI270754.1', 'KI270710.1', 'KI270717.1', 'KI270724.1', 'KI270720.1', 'KI270723.1', 'KI270718.1', 'KI270317.1', 'KI270740.1', 'KI270755.1', 'KI270707.1', 'KI270579.1', 'KI270752.1', 'KI270512.1', 'KI270322.1', 'GL000226.1', 'KI270311.1', 'KI270366.1', 'KI270511.1', 'KI270448.1', 'KI270521.1', 'KI270581.1', 'KI270582.1', 'KI270515.1', 'KI270588.1', 'KI270591.1', 'KI270522.1', 'KI270507.1', 'KI270590.1', 'KI270584.1', 'KI270320.1', 'KI270382.1', 'KI270468.1', 'KI270467.1', 'KI270362.1', 'KI270517.1', 'KI270593.1', 'KI270528.1', 'KI270587.1', 'KI270364.1', 'KI270371.1', 'KI270333.1', 'KI270374.1', 'KI270411.1', 'KI270414.1', 'KI270510.1', 'KI270390.1', 'KI270375.1', 'KI270420.1', 'KI270509.1', 'KI270315.1', 'KI270302.1', 'KI270518.1', 'KI270530.1', 'KI270304.1', 'KI270418.1', 'KI270424.1', 'KI270417.1', 'KI270508.1', 'KI270303.1', 'KI270381.1', 'KI270529.1', 'KI270425.1', 'KI270396.1', 'KI270363.1', 'KI270386.1', 'KI270465.1', 'KI270383.1', 'KI270384.1', 'KI270330.1', 'KI270372.1', 'KI270548.1', 'KI270580.1', 'KI270387.1', 'KI270391.1', 'KI270305.1', 'KI270373.1', 'KI270422.1', 'KI270316.1', 'KI270340.1', 'KI270338.1', 'KI270583.1', 'KI270334.1', 'KI270429.1', 'KI270393.1', 'KI270516.1', 'KI270389.1', 'KI270466.1', 'KI270388.1', 'KI270544.1', 'KI270310.1', 'KI270412.1', 'KI270395.1', 'KI270376.1', 'KI270337.1', 'KI270335.1', 'KI270378.1', 'KI270379.1', 'KI270329.1', 'KI270419.1', 'KI270336.1', 'KI270312.1', 'KI270539.1', 'KI270385.1', 'KI270423.1', 'KI270392.1', 'KI270394.1']
[INFO] Elapsed time: 0:00:01
[INFO] PyPGx finished
[INFO] PyPGx v0.1.22
[INFO] Command:
[INFO]     /Users/kokyriakidis/Documents/GitHub/Pharmakon/.snakemake/conda/3247b051/bin/pypgx bam2gdf hg38 gstt1 vdr /Users/kokyriakidis/Desktop/test_folder/KOSTAS/genes/gstt1/gdf/gstt1.gdf /Users/kokyriakidis/Desktop/test_folder/KOSTAS/applybqsr/KOSTAS.bam
Traceback (most recent call last):
  File "/Users/kokyriakidis/Documents/GitHub/Pharmakon/.snakemake/conda/3247b051/bin/pypgx", line 8, in <module>
    sys.exit(main())
  File "/Users/kokyriakidis/Documents/GitHub/Pharmakon/.snakemake/conda/3247b051/lib/python3.6/site-packages/pypgx/__main__.py", line 598, in main
    result = PYPGX_TOOLS[args.tool](**vars(args))
  File "/Users/kokyriakidis/Documents/GitHub/Pharmakon/.snakemake/conda/3247b051/lib/python3.6/site-packages/pypgx/common.py", line 233, in wrapper
    func(*args, **kwargs, input_files=input_files)
  File "/Users/kokyriakidis/Documents/GitHub/Pharmakon/.snakemake/conda/3247b051/lib/python3.6/site-packages/pypgx/bam2gdf.py", line 56, in bam2gdf
    sdf = bam2sdf(genome_build, target_gene, control_gene, input_files)
  File "/Users/kokyriakidis/Documents/GitHub/Pharmakon/.snakemake/conda/3247b051/lib/python3.6/site-packages/pypgx/bam2sdf.py", line 51, in bam2sdf
    regions = sort_regions([tr, cr])
  File "/Users/kokyriakidis/Documents/GitHub/Pharmakon/.snakemake/conda/3247b051/lib/python3.6/site-packages/pypgx/sglib.py", line 578, in sort_regions
    return sorted(regions, key = f)
  File "/Users/kokyriakidis/Documents/GitHub/Pharmakon/.snakemake/conda/3247b051/lib/python3.6/site-packages/pypgx/sglib.py", line 576, in f
    chr = int(r[0].replace("chr", ""))
ValueError: invalid literal for int() with base 10: '22_KI270879v1_alt'
```
I am using GRCh38 reference genome from Ensemble (release 100)
Any clue?

**Comment by sbslee** [comment]
I think the problem is that the GSTT1 gene is located on an alternative contig of chr22 in hg38 (chr22_KI270879v1_alt). I will take a look at this, but please note that Stargazer currently does not support genotyping of the gene. Please read [this](https://github.com/sbslee/stargazer/issues/1) for details.

**Comment by sbslee** [comment]
Hi, I was able to replicate the issue. It was indeed caused by the fact that GSTT1 has an unusual congtig name for hg38 (``chr22_KI270879v1_alt ``). This was fixed in v0.1.23 (now available via PyPi). See below for working example.

```
[sbslee@grchead ~]$ pypgx bam2gdf hg38 gstt1 vdr test2.gdf /nfs/home/sbslee/projects/20200518_cdc_hg19_to_hg38_test/stargazer/cdcpgx.stargazer-pipeline-remap.project/bam/74279.sorted.markeddups.recal.bam
[INFO] PyPGx v0.1.23
[INFO] Command:
[INFO]     /nfs/home/sbslee/.local/bin/pypgx bam2gdf hg38 gstt1 vdr test2.gdf /nfs/home/sbslee/projects/20200518_cdc_hg19_to_hg38_test/stargazer/cdcpgx.stargazer-pipeline-remap.project/bam/74279.sorted.markeddups.recal.bam
[INFO] Sample IDs: ['74279']
[INFO] Contigs: ['chr1', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', ..., DRB1*15:01:01:02', 'HLA-DRB1*15:01:01:03', 'HLA-DRB1*15:01:01:04', 'HLA-DRB1*15:02:01', 'HLA-DRB1*15:03:01:01', 'HLA-DRB1*15:03:01:02', 'HLA-DRB1*16:02:01']
[INFO] Elapsed time: 0:00:04
[INFO] PyPGx finished
```

P.S. I just noticed that this fix won’t solve your problem, unfortunately. It looks like your BAM file does not have the ``chr22_KI270879v1_alt `` contig at all. It probably means that the reference FASTA file used to create your BAM file did not have that contig.

I will close this for now because the original issue was resolved. Please re-open it if you need more support.
